### PR TITLE
Add vsock_perf test case

### PR DESCRIPTION
**Pull request by qcheng-redhat**
Add vsock_perf test case

ID: 2122
Signed-off-by: qcheng-redhat <qcheng@redhat.com>

### CODE REVIEW

**Comment by qcheng-redhat** [comment]
(1/1) Host_RHEL.m9.u5.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.5.0.x86_64.io-github-autotest-qemu.vsock_perf.q35: STARTED
 (1/1) Host_RHEL.m9.u5.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.5.0.x86_64.io-github-autotest-qemu.vsock_perf.q35: PASS (207.24 s)

**Comment by qcheng-redhat** [comment]
Hi @XueqiangWei, can you help review this patch, when you have time.

**Comment by qcheng-redhat** [comment]
hi @BohdanMar , if you need add vsock performance test on your platform, please help review this patch. Thanks.

**Comment by qcheng-redhat** [comment]
Hi @PaulYuuu, please help review this patch, when you have time. Thanks

File: qemu/tests/vsock_perf.py [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+    """
+    vm = env.get_vm(params["main_vm"])
+    vm.verify_alive()
+    session = vm.wait_for_login()
+
+    disable_firewall = params.get("disable_firewall")
+    session.cmd(disable_firewall, ignore_all_errors=True)
+
+    kernel_version = process.getoutput("uname -r", shell=True)
+    test.log.info("Current kernel version: %s" % kernel_version)
```
[the remark this answers is not in the document]
Comment (qcheng-redhat) [reply qemu/tests/vsock_perf.py]: Removed the log info here.

File: qemu/tests/vsock_perf.py [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+    :param env: Dictionary with test environment
+    """
+    vm = env.get_vm(params["main_vm"])
+    vm.verify_alive()
+    session = vm.wait_for_login()
+
+    disable_firewall = params.get("disable_firewall")
+    session.cmd(disable_firewall, ignore_all_errors=True)
+
+    kernel_version = process.getoutput("uname -r", shell=True)
```
[the remark this answers is not in the document]
Comment (qcheng-redhat) [reply qemu/tests/vsock_perf.py]: Thanks. Updated.

File: qemu/tests/vsock_perf.py [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+    vsock_test_base_dir = "/home/"
+    vsock_test_src_file = os.path.join(
+        data_dir.get_deps_dir("vsock_test"),
+        "vsock_test.tar.xz")
+
+    rm_cmd = "rm -rf %s" % os.path.join(vsock_test_base_dir, "vsock*")
+    process.system(rm_cmd, shell=True, ignore_status=True)
+    session.cmd(rm_cmd, ignore_all_errors=True)
+    cp_cmd = "cp %s %s" % (vsock_test_src_file, vsock_test_base_dir)
+    process.system(cp_cmd, shell=True)
```
[the remark this answers is not in the document]
Comment (qcheng-redhat) [reply qemu/tests/vsock_perf.py]: Thanks. Updated.

File: qemu/tests/vsock_perf.py [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+    if (host_status or guest_status) != 0:
+        process.system(rm_cmd, shell=True, ignore_status=True)
+        session.cmd_output_safe(rm_cmd, ignore_all_errors=True)
+        session.close()
+        test.error("Complile failed")
+    return os.path.join(vsock_test_base_dir, "vsock/vsock_perf")
+
+
+def run_host_cmd(host_cmd, host_file):
+    status, host_output = process.getstatusoutput(
```
Comment (PaulYuuu) [inline qemu/tests/vsock_perf.py:48]: `status` is not used, suggest `process.getoutput`

File: qemu/tests/vsock_perf.py [hunk 5]
```diff
@@ -0,0 +1,111 @@
...
+        host_cmd, timeout=30, shell=True)
+    time.sleep(3)
```
Comment (PaulYuuu) [inline qemu/tests/vsock_perf.py:50]: Any reason sleep here?

File: qemu/tests/vsock_perf.py [hunk 6]
```diff
@@ -0,0 +1,111 @@
...
+        test_bin = copy_compile_testsuite(test, vm, session)
+        test.log.info("test_bin: %s" % test_bin)
+        host_file = params["host_file"] + kernel_version
+        guest_file = params["guest_file"] + kernel_version
+
+        host_cmd = params["host_cmd"]
+        guest_cmd = params["guest_cmd"]
+        for _ in range(params.get_numeric("repeate_times")):
+            test.log.info("Start perf test from host ...")
+            host_thread = threading.Thread(
```
Comment (PaulYuuu) [inline qemu/tests/vsock_perf.py:90]: `client.run_bg`

File: qemu/tests/vsock_perf.py [hunk 7]
```diff
@@ -0,0 +1,111 @@
...
+            host_thread.start()
+            time.sleep(5)
+            test.log.info("Start perf test from guest ...")
+            status, guest_output = session.cmd_status_output(guest_cmd)
+            with open(guest_file, 'a') as file:
+                file.write(guest_output + '\n')
+            host_thread.join()
+            test.log.info("loop number %s" % _)
+            time.sleep(10)
+            if status != 0:
```
Comment (PaulYuuu) [inline qemu/tests/vsock_perf.py:102]: Any reason for some `sleep`?

**Review by BohdanMar — APPROVED** [review]
I guess, for now, this test case is not necessary to implement on aarch64 platform.

File: qemu/tests/cfg/vsock_perf.cfg [hunk 8]
```diff
@@ -0,0 +1,14 @@
+- vsock_perf:
+    no Windows
+    no RHEL.6 RHEL.7 RHEL.8
+    no Host_RHEL.m6 Host_RHEL.m7 Host_RHEL.m8
+    type = vsock_perf
+    vsocks = vhost_vsock0
+    only q35
```
[the remark this answers is not in the document]
Comment (qcheng-redhat) [reply qemu/tests/cfg/vsock_perf.cfg:7]: @BohdanMar Thanks for review. I will not cover aarch64 in this patch.

File: qemu/tests/vsock_perf.py (hunk 5 above)
Comment (qcheng-redhat) [reply qemu/tests/vsock_perf.py]: Removed this part.

File: qemu/tests/vsock_perf.py (hunk 6 above)
Comment (qcheng-redhat) [reply qemu/tests/vsock_perf.py]: Changed to use `run_bg`

File: qemu/tests/vsock_perf.py (hunk 7 above)
Comment (qcheng-redhat) [reply qemu/tests/vsock_perf.py]: Removed.

File: qemu/tests/vsock_perf.py [hunk 9]
```diff
@@ -0,0 +1,100 @@
...
+        vsock_test_base_dir, "vsock_test.tar.xz")
+    process.system(uncompress_cmd, shell=True, ignore_status=True)
+    session.cmd(uncompress_cmd)
+    compile_cmd = "cd %s && make vsock_perf" % (os.path.join(
+        vsock_test_base_dir, "vsock/"))
+    host_status = process.system(compile_cmd, shell=True)
+    guest_status = session.cmd_status(compile_cmd)
+    if host_status or guest_status:
+        process.system(rm_cmd, shell=True, ignore_status=True)
+        session.cmd_output_safe(rm_cmd, ignore_all_errors=True)
```
Comment (PaulYuuu) [inline qemu/tests/vsock_perf.py:42]: Covered in the `finally` part?

File: qemu/tests/vsock_perf.py [hunk 10]
```diff
@@ -0,0 +1,100 @@
...
+                file.write(host_output.get_output() + '\n')
+            with open(guest_file, 'a') as file:
+                file.write(guest_output + '\n')
+            test.log.info("loop number %s" % _)
+
+            if host_output.get_status() or guest_status:
+                test.fail("vsock_perf test failed")
+
+    finally:
+        rm_cmd = "rm -rf /home/vsock*"
```
[the remark this answers is not in the document]
Comment (qcheng-redhat) [reply qemu/tests/vsock_perf.py]: updated.

File: qemu/tests/vsock_perf.py (hunk 9 above)
Comment (qcheng-redhat) [reply qemu/tests/vsock_perf.py]: updated.

File: qemu/tests/cfg/vsock_perf.cfg (hunk 8 above)
[the remark this answers is not in the document]
Comment (PaulYuuu) [reply qemu/tests/cfg/vsock_perf.cfg:7]: So the case can work on other platforms, why diable it for upstream users?

[the remark this answers is not in the document]
Comment (qcheng-redhat) [reply qemu/tests/cfg/vsock_perf.cfg]: Updated. Also works on other architectures.

File: qemu/tests/vsock_perf.py [hunk 11]
```diff
@@ -0,0 +1,98 @@
...
+    vm.verify_alive()
+    session = vm.wait_for_login()
+
+    disable_firewall = params.get("disable_firewall")
+    session.cmd(disable_firewall, ignore_all_errors=True)
+
+    kernel_version = platform.uname().release
+    vsock_test_base_dir = "/home/"
+
+    rm_cmd = "rm -rf %s" % os.path.join(vsock_test_base_dir, "vsock*")
```
Comment (PaulYuuu) [inline qemu/tests/vsock_perf.py:69]: Not used.

File: qemu/tests/vsock_perf.py [hunk 12]
```diff
@@ -0,0 +1,98 @@
...
+        test.log.info("test_bin: %s" % test_bin)
+
+        host_file = params["host_file"] + kernel_version
+        guest_file = params["guest_file"] + kernel_version
+        host_cmd = params["host_cmd"]
+        guest_cmd = params["guest_cmd"]
+
+        for _ in range(params.get_numeric("repeate_times")):
+            test.log.info("Start perf test from host ...")
+            host_output = aexpect.run_bg(host_cmd, timeout=30)
```
Comment (PaulYuuu) [inline qemu/tests/vsock_perf.py:82]: I am not sure when the process will stop, but before the next run time, ensure the current process is closed would be good.

Comment (qcheng-redhat) [reply qemu/tests/vsock_perf.py:80]: Updated.

**Review by PaulYuuu — APPROVED** [review]

**Comment by PaulYuuu** [comment]
Latest result please.

**Comment by qcheng-redhat** [comment]
> Latest result please.

 (1/1) Host_RHEL.m9.u5.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.5.0.x86_64.io-github-autotest-qemu.vsock_perf.q35: STARTED
 (1/1) Host_RHEL.m9.u5.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.5.0.x86_64.io-github-autotest-qemu.vsock_perf.q35: PASS (299.25 s)

**Pull request merged**